activerecord/lib/active_record/reflection.rb

```diff
@@ -877,5 +897,81 @@ def derive_class_name
         delegate(*delegate_methods, to: :delegate_reflection)
 
     end
+
+    class PolymorphicReflection < ThroughReflection # :nodoc:
+      def initialize(reflection, prev_reflection)
+        @reflection = reflection
+        @prev_reflection = prev_reflection
+      end
+
+      def klass
+        @reflection.klass
+      end
+
+      def scope
+        @reflection.scope
+      end
+
+      def table_name
+        @reflection.table_name
+      end
+
+      def plural_name
+        @reflection.plural_name
+      end
+
+      def join_keys(assoc_klass)
```

Comment:
Should I change the others too? If I change this one it won't match the other join_keys methods in the reflection class like https://github.com/rails/rails/blob/master/activerecord/lib/active_record/reflection.rb#L152